topydo/ui/columns/CommandLineWidget.py

```diff
@@ -71,12 +83,123 @@ def _history_back(self):
         if self.history_pos != 0:
             self._history_move(-1)
 
+    def insert_completion(self, p_insert):
+        start, end = self._surrounding_text
+        final_text = start + p_insert + end
+        self.set_edit_text(final_text)
+        self.set_edit_pos(len(start) + len(p_insert))
+
+    @property
+    def completion_mode(self):
+        return len(self.completion_box) > 1
+
+    @completion_mode.setter
+    def completion_mode(self, p_enable):
+        if p_enable is True:
+            urwid.emit_signal(self, 'show_completions')
+        elif p_enable is False:
+            self._surrounding_text = None
+            if self.completion_mode:
+                self.completion_box.clear()
+                urwid.emit_signal(self, 'hide_completions')
+
+    def _complete(self):
+        def find_word_start(p_text, p_pos):
+            """ Returns position of the beginning of a word ending in p_pos. """
+            return p_text.lstrip().rfind(' ', 0, p_pos) + 1
+
+        def get_word_before_pos(p_text, p_pos):
+            start = find_word_start(p_text, p_pos)
+
+            return (p_text[start:p_pos], start)
+
+        pos = self.edit_pos
+        text = self.edit_text
+        completer = self.completer
+
+        word_before_cursor, start = get_word_before_pos(text, pos)
+        completions = completer.get_completions(word_before_cursor, start == 0)
+        # store slices before and after place for completion
+        self._surrounding_text = (text[:start], text[pos:])
+        single_completion = len(completions) == 1
+
+        if (single_completion and completions[0] == word_before_cursor
```

Comment:
They're added according to [PEP8 suggestion](https://www.python.org/dev/peps/pep-0008/#should-a-line-break-before-or-after-a-binary-operator)